src/main/java/executor/service/service/scenarios/impl/ScenarioSourceListenerImpl.java

```diff
@@ -0,0 +1,112 @@
+package executor.service.service.scenarios.impl;
+
+import executor.service.handler.ScenarioHandler;
+import executor.service.handler.TerminatorListener;
+import executor.service.model.scenario.Scenario;
+import executor.service.service.scenarios.ScenarioProvider;
+import executor.service.service.scenarios.ScenarioSourceListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * The {@code ScenarioSourceListenerImpl} class is an implementation of the {@link ScenarioSourceListener}
+ * interface. It listens for scenarios from a {@link ScenarioProvider} and processes them using a provided
+ * {@link ScenarioHandler}. The scenarios are continuously processed in a loop with a specified delay.
+ *  * </pre>
+ *  * Alternatively, you can use the overloaded execute method with a {@link TerminatorListener} to control
+ *  * the termination of the listening process.
+ *  * <pre>
+ * @author Yurii Kotsiuba
+ * @see ScenarioSourceListener
+ * @see ScenarioProvider
+ * @see ScenarioHandler
+ */
+public class ScenarioSourceListenerImpl implements ScenarioSourceListener {
+
+    private static final long DELAY = 1000;
```

Comment:
In your code, the delay between iterations is set to 1000 milliseconds (1 second) and cannot be changed. Consider adding a configuration that allows users to set their own delay if they so desire.